source/_integrations/modbus.markdown

```diff
@@ -1095,70 +1378,170 @@ modbus:
+          description: "Value when switch is on."
           required: false
-          default: same as command_on
+          default: "Same as `command_on`"
           type: integer
         state_off:
-          description: value when switch is off.
+          description: "Value when switch is off."
           required: false
-          default: same as command_off
+          default: "Same as `command_off`"
           type: integer
-    unique_id:
-      description: An ID that uniquely identifies this sensor. If two sensors have the same unique ID, Home Assistant will raise an exception.
-      required: false
-      type: string
+
 {% endconfiguration %}
 
-## Opening an issue
+### Example: switch configuration
+
+```yaml
+# Example configuration.yaml entry
+modbus:
+  - type: tcp
+    host: IP_ADDRESS
+    port: 502
+    switches:
+      - name: Switch1
+        address: 13
+        write_type: coil
+      - name: Switch2
+        slave: 2
+        address: 14
+        write_type: coil
+        verify:
+      - name: Register1
+        address: 11
+        command_on: 1
+        command_off: 0
+        verify:
+            input_type: holding
+            address: 127
+            state_on: 25
+            state_off: 1
+```
+
+# modbus services
+
+The modbus integration provides two generic write services in addition to the platform-specific services.
+
+| Service | Description |
+| ------- | ----------- |
+| modbus.write_register | Write register or registers |
+| modbus.write_coil | Write coil or coils |
+
+Description:
+
+| Attribute | Description |
+| --------- | ----------- |
+| hub       | Hub name (defaults to 'modbus_hub' when omitted) |
+| unit      | Slave address (0-255), alternative to slave |
+| slave     | Slave address (0-255), alternative to unit |
+| address   | Address of the Register (e.g. 138) |
+| value     | (write_register) A single value or an array of 16-bit values. Single value will call modbus function code 0x06. Array will call modbus function code 0x10. Values might need reverse ordering. E.g., to set 0x0004 you might need to set `[4,0]`, this depend on the byte order of your CPU |
+| state     | (write_coil) A single boolean or an array of booleans. Single boolean will call modbus function code 0x05. Array will call modbus function code 0x0F |
+
+The modbus integration also provides communication stop/restart services. These services will not do any reconfiguring, but simply stop/start the modbus communication layer.
+
+| Service | Description |
+| ------- | ----------- |
+| modbus.stop | Stop communication |
+| modbus.restart | Restart communication (Stop first if running) |
+
+Description:
+
+| Attribute | Description |
+| --------- | ----------- |
+| hub       | Hub name (defaults to 'modbus_hub' when omitted) |
+
+## Example: writing a float32 type register
+
+To write a float32 datatype register use network format like `10.0` == `0x41200000` (network order float hexadecimal).
```

Comment:
```suggestion
To write a float32 datatype register, use a network format like `10.0` == `0x41200000` (network order float hexadecimal).
```

Reply:
"a" indicated "one of of several" which is not the case, there are only 1 network format.